packages/gatsby/cache-dir/navigation.js

```diff
@@ -222,6 +216,12 @@ class RouteUpdates extends React.Component {
     return false
   }
 
+  componentDidUpdate(prevProps) {
+    if (this.props.location.pathname !== prevProps.location.pathname) {
```

Comment:
Does this break when other than `pathname` parts of location are changing? I.e. `search`, as a result causing problems like #27261

Reply:
Good spot! but I do find it strange how this can cause problems like #27261 because the comparison has not changed.

I removed `getSnapshotBeforeUpdate` which returned a boolean based on the same condition `this.props.location.pathname !== prevProps.location.pathname`.

That boolean was then passed into `componentDidUpdate ` as the third parameter which is what was being used as the condition just like what I am doing here.

So I am not sure I understand why this wasn't an issue before, regardless - I am happy to investigate this 👍